File: appthemehelper/src/main/java/code/name/monkey/appthemehelper/util/TabLayoutUtil.java
package code.name.monkey.appthemehelper.util;

import android.content.res.ColorStateList;
import androidx.annotation.ColorInt;
import androidx.annotation.Nullable;
import com.google.android.material.tabs.TabLayout;


public final class TabLayoutUtil {

    public static void setTabIconColors(@Nullable TabLayout tabLayout, @ColorInt int normalColor, @ColorInt int selectedColor) {
        if (tabLayout == null)
            return;

        final ColorStateList sl = new ColorStateList(new int[][]{
                new int[]{-android.R.attr.state_selected},
                new int[]{android.R.attr.state_selected}
        },
                new int[]{
                        normalColor,
                        selectedColor
                });
        for (int i = 0; i < tabLayout.getTabCount(); i++) {
            final TabLayout.Tab tab = tabLayout.getTabAt(i);
            if (tab != null && tab.getIcon() != null) {
                tab.setIcon(TintHelper.createTintedDrawable(tab.getIcon(), sl));
            }
        }
    }

    private TabLayoutUtil() {
    }
}

File: appthemehelper/src/main/java/code/name/monkey/appthemehelper/util/TabLayoutUtil.kt
package code.name.monkey.appthemehelper.util

import android.content.res.ColorStateList
import androidx.annotation.ColorInt
import com.google.android.material.tabs.TabLayout


object TabLayoutUtil {

    fun setTabIconColors(tabLayout: TabLayout?, @ColorInt normalColor: Int, @ColorInt selectedColor: Int) {
        if (tabLayout == null)
            return

        val sl = ColorStateList(arrayOf(intArrayOf(-android.R.attr.state_selected), intArrayOf(android.R.attr.state_selected)),
                intArrayOf(normalColor, selectedColor))
        for (i in 0 until tabLayout.tabCount) {
            val tab = tabLayout.getTabAt(i)
            if (tab != null && tab.icon != null) {
                tab.icon = TintHelper.createTintedDrawable(tab.icon, sl)
            }
        }
    }
}
